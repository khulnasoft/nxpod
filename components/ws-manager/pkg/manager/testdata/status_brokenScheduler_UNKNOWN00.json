{
    "pod": {
        "metadata": {
            "name": "ws-fb291b69-7bbc-4865-a432-33f558f20091",
            "namespace": "staging-cw-scheduler",
            "selfLink": "/api/v1/namespaces/staging-cw-scheduler/pods/ws-fb291b69-7bbc-4865-a432-33f558f20091",
            "uid": "283aff69-8dd8-11e9-97df-42010a8402a0",
            "resourceVersion": "108100474",
            "creationTimestamp": "2019-06-13T12:38:33Z",
            "labels": {
                "app": "nxpod",
                "component": "workspace",
                "nxpod.khulnasoft.com/networkpolicy": "default",
                "gpwsman": "true",
                "headless": "false",
                "metaID": "a9949b21-d6d8-443a-85f5-44f5ac3065c0",
                "owner": "34032a87-deb7-46f3-9044-0c038679c822",
                "workspaceID": "fb291b69-7bbc-4865-a432-33f558f20091",
                "workspaceType": "regular"
            },
            "annotations": {
                "nxpod/id": "fb291b69-7bbc-4865-a432-33f558f20091",
                "nxpod/servicePrefix": "a9949b21-d6d8-443a-85f5-44f5ac3065c0",
                "nxpod/traceid": "AAAAAAAAAAC0nYySXpfHcnzoa970pCkZBtBSaqGr/6oBAAAAAA==",
                "nxpod/url": "http://a9949b21-d6d8-443a-85f5-44f5ac3065c0.ws-eu.cw-scheduler.staging.nxpod.khulnasoft.com",
                "nxpod/never-ready": "true",
                "prometheus.io/path": "/metrics",
                "prometheus.io/port": "23000",
                "prometheus.io/scrape": "true"
            }
        },
        "spec": {
            "volumes": [
                {
                    "name": "vol-this-theia",
                    "hostPath": {
                        "path": "/mnt/disks/ssd0/theia/theia-cw-scheduler.17",
                        "type": "Directory"
                    }
                },
                {
                    "name": "vol-this-workspace",
                    "hostPath": {
                        "path": "/mnt/disks/ssd0/workspaces/fb291b69-7bbc-4865-a432-33f558f20091",
                        "type": "DirectoryOrCreate"
                    }
                },
                {
                    "name": "vol-sync-tmp",
                    "hostPath": {
                        "path": "/mnt/disks/ssd0/workspaces/sync-tmp",
                        "type": "DirectoryOrCreate"
                    }
                }
            ],
            "containers": [
                {
                    "name": "workspace",
                    "image": "eu.gcr.io/nxpod-dev/workspace-images/b24b4698d67f493b801d17196870fd8a422ffa1e/eu.gcr.io/nxpod-dev/workspace-full:sha256-4bcfae6785bcfa55f3fd693af74b3ceeb032e26c99c60b64bc9fe06980efcd23",
                    "ports": [
                        {
                            "containerPort": 23000,
                            "protocol": "TCP"
                        }
                    ],
                    "env": [
                        {
                            "name": "NXPOD_REPO_ROOT",
                            "value": "/workspace/bel"
                        },
                        {
                            "name": "NXPOD_CLI_APITOKEN",
                            "value": "f24b79c3-8ea9-4638-b2bb-8b2fef89e9c8"
                        },
                        {
                            "name": "NXPOD_WORKSPACE_ID",
                            "value": "a9949b21-d6d8-443a-85f5-44f5ac3065c0"
                        },
                        {
                            "name": "NXPOD_INSTANCE_ID",
                            "value": "fb291b69-7bbc-4865-a432-33f558f20091"
                        },
                        {
                            "name": "NXPOD_GIT_USER_NAME",
                            "value": "Christian Weichel"
                        },
                        {
                            "name": "NXPOD_GIT_USER_EMAIL",
                            "value": "some@user.com"
                        },
                        {
                            "name": "NXPOD_THEIA_PORT",
                            "value": "23000"
                        },
                        {
                            "name": "THEIA_WORKSPACE_ROOT",
                            "value": "/workspace"
                        },
                        {
                            "name": "NXPOD_HOST",
                            "value": "http://cw-scheduler.staging.nxpod.khulnasoft.com"
                        },
                        {
                            "name": "NXPOD_WSSYNC_APITOKEN",
                            "value": "c58ded78-688a-47d4-bd90-f398d794452b"
                        },
                        {
                            "name": "NXPOD_WSSYNC_APIPORT",
                            "value": "44444"
                        },
                        {
                            "name": "NXPOD_WORKSPACE_URL",
                            "value": "http://a9949b21-d6d8-443a-85f5-44f5ac3065c0.ws-eu.cw-scheduler.staging.nxpod.khulnasoft.com"
                        },
                        {
                            "name": "USER_ENV_NXPOD_TASKS",
                            "value": "[{\"init\":\"cd /workspace/bel && go get -v && go test -v ./...\",\"command\":\"cd /workspace/bel && go run examples/*\"}]"
                        },
                        {
                            "name": "NXPOD_INTERVAL",
                            "value": "30000"
                        },
                        {
                            "name": "NXPOD_MEMORY",
                            "value": "3403"
                        },
                        {
                            "name": "NXPOD_TASKS",
                            "value": "[{\"init\":\"cd /workspace/bel && go get -v && go test -v ./...\",\"command\":\"cd /workspace/bel && go run examples/*\"}]"
                        }
                    ],
                    "resources": {
                        "limits": {
                            "cpu": "7",
                            "memory": "8366Mi"
                        },
                        "requests": {
                            "cpu": "1m",
                            "memory": "3246Mi"
                        }
                    },
                    "volumeMounts": [
                        {
                            "name": "vol-this-workspace",
                            "mountPath": "/workspace"
                        },
                        {
                            "name": "vol-this-theia",
                            "readOnly": true,
                            "mountPath": "/theia"
                        }
                    ],
                    "livenessProbe": {
                        "httpGet": {
                            "path": "/",
                            "port": 23000,
                            "scheme": "HTTP"
                        },
                        "timeoutSeconds": 1,
                        "periodSeconds": 30,
                        "successThreshold": 1,
                        "failureThreshold": 3
                    },
                    "readinessProbe": {
                        "httpGet": {
                            "path": "/",
                            "port": 23000,
                            "scheme": "HTTP"
                        },
                        "timeoutSeconds": 1,
                        "periodSeconds": 1,
                        "successThreshold": 1,
                        "failureThreshold": 600
                    },
                    "terminationMessagePath": "/dev/termination-log",
                    "terminationMessagePolicy": "File",
                    "imagePullPolicy": "Always",
                    "securityContext": {
                        "capabilities": {
                            "add": [
                                "AUDIT_WRITE",
                                "FSETID",
                                "KILL",
                                "NET_BIND_SERVICE"
                            ],
                            "drop": [
                                "SETPCAP",
                                "CHOWN",
                                "NET_RAW",
                                "DAC_OVERRIDE",
                                "FOWNER",
                                "SYS_CHROOT",
                                "SETFCAP",
                                "SETUID",
                                "SETGID"
                            ]
                        },
                        "privileged": false,
                        "runAsUser": 33333,
                        "runAsNonRoot": true,
                        "readOnlyRootFilesystem": false,
                        "allowPrivilegeEscalation": false
                    }
                }
            ],
            "restartPolicy": "Always",
            "terminationGracePeriodSeconds": 30,
            "dnsPolicy": "None",
            "serviceAccountName": "workspace",
            "serviceAccount": "workspace",
            "automountServiceAccountToken": false,
            "nodeName": "gke-nxpod-dev-default-pool-b25aa276-3tl2",
            "securityContext": {},
            "imagePullSecrets": [
                {
                    "name": "dockerhub-typefox"
                },
                {
                    "name": "eu.gcr.io-nxpod"
                }
            ],
            "affinity": {
                "nodeAffinity": {
                    "requiredDuringSchedulingIgnoredDuringExecution": {
                        "nodeSelectorTerms": [
                            {
                                "matchExpressions": [
                                    {
                                        "key": "nxpod.khulnasoft.com/workload_workspace",
                                        "operator": "In",
                                        "values": [
                                            "true"
                                        ]
                                    }
                                ]
                            }
                        ]
                    }
                }
            },
            "schedulerName": "workspace-scheduler",
            "tolerations": [
                {
                    "key": "node.kubernetes.io/not-ready",
                    "operator": "Exists",
                    "effect": "NoExecute",
                    "tolerationSeconds": 300
                },
                {
                    "key": "node.kubernetes.io/unreachable",
                    "operator": "Exists",
                    "effect": "NoExecute",
                    "tolerationSeconds": 300
                }
            ],
            "priority": 0,
            "dnsConfig": {
                "nameservers": [
                    "1.1.1.1",
                    "8.8.8.8"
                ]
            }
        },
        "status": {
            "phase": "Failed",
            "message": "Pod Predicate MatchNodeSelector failed",
            "reason": "MatchNodeSelector",
            "startTime": "2019-06-13T12:41:05Z"
        }
    },
    "theiaService": {
        "metadata": {
            "name": "ws-a9949b21-d6d8-443a-85f5-44f5ac3065c0-theia",
            "namespace": "staging-cw-scheduler",
            "selfLink": "/api/v1/namespaces/staging-cw-scheduler/services/ws-a9949b21-d6d8-443a-85f5-44f5ac3065c0-theia",
            "uid": "283c74be-8dd8-11e9-97df-42010a8402a0",
            "resourceVersion": "108099691",
            "creationTimestamp": "2019-06-13T12:38:33Z",
            "labels": {
                "app": "nxpod",
                "component": "workspace",
                "gpwsman": "true",
                "headless": "false",
                "metaId": "a9949b21-d6d8-443a-85f5-44f5ac3065c0",
                "owner": "34032a87-deb7-46f3-9044-0c038679c822",
                "workspaceID": "fb291b69-7bbc-4865-a432-33f558f20091"
            }
        },
        "spec": {
            "ports": [
                {
                    "name": "theia",
                    "protocol": "TCP",
                    "port": 23000,
                    "targetPort": 23000
                }
            ],
            "selector": {
                "app": "nxpod",
                "component": "workspace",
                "gpwsman": "true",
                "headless": "false",
                "metaId": "a9949b21-d6d8-443a-85f5-44f5ac3065c0",
                "owner": "34032a87-deb7-46f3-9044-0c038679c822",
                "workspaceID": "fb291b69-7bbc-4865-a432-33f558f20091"
            },
            "clusterIP": "10.3.243.187",
            "type": "ClusterIP",
            "sessionAffinity": "None"
        },
        "status": {
            "loadBalancer": {}
        }
    },
    "events": [
        {
            "metadata": {
                "name": "ws-fb291b69-7bbc-4865-a432-33f558f20091-r6w9p",
                "generateName": "ws-fb291b69-7bbc-4865-a432-33f558f20091-",
                "namespace": "staging-cw-scheduler",
                "selfLink": "/api/v1/namespaces/staging-cw-scheduler/events/ws-fb291b69-7bbc-4865-a432-33f558f20091-r6w9p",
                "uid": "82abbaff-8dd8-11e9-97df-42010a8402a0",
                "resourceVersion": "3452265",
                "creationTimestamp": "2019-06-13T12:41:05Z"
            },
            "involvedObject": {
                "kind": "Pod",
                "namespace": "staging-cw-scheduler",
                "name": "ws-fb291b69-7bbc-4865-a432-33f558f20091",
                "uid": "283aff69-8dd8-11e9-97df-42010a8402a0"
            },
            "reason": "Scheduled",
            "message": "Placed pod [staging-cw-scheduler/ws-fb291b69-7bbc-4865-a432-33f558f20091] on gke-nxpod-dev-default-pool-b25aa276-3tl2\n",
            "source": {
                "component": "workspace-scheduler"
            },
            "firstTimestamp": "2019-06-13T12:41:05Z",
            "lastTimestamp": "2019-06-13T12:41:05Z",
            "count": 1,
            "type": "Normal",
            "eventTime": null,
            "reportingComponent": "",
            "reportingInstance": ""
        },
        {
            "metadata": {
                "name": "ws-fb291b69-7bbc-4865-a432-33f558f20091.15a7c2836f2b2e6f",
                "namespace": "staging-cw-scheduler",
                "selfLink": "/api/v1/namespaces/staging-cw-scheduler/events/ws-fb291b69-7bbc-4865-a432-33f558f20091.15a7c2836f2b2e6f",
                "uid": "82ad4d41-8dd8-11e9-97df-42010a8402a0",
                "resourceVersion": "3452267",
                "creationTimestamp": "2019-06-13T12:41:05Z"
            },
            "involvedObject": {
                "kind": "Pod",
                "namespace": "staging-cw-scheduler",
                "name": "ws-fb291b69-7bbc-4865-a432-33f558f20091",
                "uid": "283aff69-8dd8-11e9-97df-42010a8402a0",
                "apiVersion": "v1",
                "resourceVersion": "108100472"
            },
            "reason": "MatchNodeSelector",
            "message": "Predicate MatchNodeSelector failed",
            "source": {
                "component": "kubelet",
                "host": "gke-nxpod-dev-default-pool-b25aa276-3tl2"
            },
            "firstTimestamp": "2019-06-13T12:41:05Z",
            "lastTimestamp": "2019-06-13T12:41:05Z",
            "count": 1,
            "type": "Warning",
            "eventTime": null,
            "reportingComponent": "",
            "reportingInstance": ""
        }
    ]
}
