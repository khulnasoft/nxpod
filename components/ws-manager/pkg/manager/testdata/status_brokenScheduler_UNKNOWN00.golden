{
    "status": {
        "id": "fb291b69-7bbc-4865-a432-33f558f20091",
        "metadata": {
            "owner": "34032a87-deb7-46f3-9044-0c038679c822",
            "meta_id": "a9949b21-d6d8-443a-85f5-44f5ac3065c0",
            "started_at": {
                "seconds": 1560429513
            }
        },
        "spec": {
            "workspace_image": "eu.gcr.io/nxpod-dev/workspace-images/b24b4698d67f493b801d17196870fd8a422ffa1e/eu.gcr.io/nxpod-dev/workspace-full:sha256-4bcfae6785bcfa55f3fd693af74b3ceeb032e26c99c60b64bc9fe06980efcd23",
            "url": "http://a9949b21-d6d8-443a-85f5-44f5ac3065c0.ws-eu.cw-scheduler.staging.nxpod.io"
        },
        "conditions": {
            "failed": "MatchNodeSelector: Pod Predicate MatchNodeSelector failed",
            "service_exists": 1,
            "deployed": 1
        },
        "runtime": {
            "node_name": "gke-nxpod-dev-default-pool-b25aa276-3tl2"
        },
        "auth": {}
    }
}